src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java

```diff
@@ -0,0 +1,224 @@
+import com.ibm.as400.data.XmlException;
+import javax.xml.stream.XMLStreamException;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.SynapseLog;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.mediation.registry.WSO2Registry;
+import org.wso2.carbon.registry.core.Resource;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A connector component that calls an AS400 program using PCML.
+ */
+public class AS400CallProgram extends AbstractConnector {
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     *     Calls a program in the AS400 server using PCML. The input parameters are taken through the soap body of the
+     *     message context.
+     * </p>
+     */
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        SynapseLog log = getLog(messageContext);
+        AS400 as400 = null;
+        try {
+            Object as400InstanceProperty = messageContext.getProperty(AS400Constants.AS400_INSTANCE);
+            if (null != as400InstanceProperty) {
+                as400 = (AS400) as400InstanceProperty;
+            } else {
+                throw new AS400PCMLConnectorException("Unable to find an AS400 instance to call program. Use the " +
+                        "'init' mediator to create an AS400 instance.");
+            }
+
+            Object pcmlFileLocationParameter = getParameter(messageContext, AS400Constants.AS400_PCML_FILE_LOCATION);
+            if (null == pcmlFileLocationParameter) {
+                throw new AS400PCMLConnectorException("A PCML file name could not be found as a parameter to call a " +
+                        "program. Make sure the registry path is correct.");
+            }
+
+            // Get PCML source file name
+            String pcmlFileLocation = (String) pcmlFileLocationParameter;
+
+            Object pcmlProgramNameParameter = getParameter(messageContext, AS400Constants.AS400_PCML_PROGRAM_NAME);
+            if (null == pcmlProgramNameParameter) {
+                throw new AS400PCMLConnectorException("A program name was not specified to call.");
+            }
+
+            // Get program name to call
+            String programName = (String) pcmlProgramNameParameter;
+
+            // Create program document by getting the PCML file from registry.
+            WSO2Registry registry = (WSO2Registry) messageContext.getConfiguration().getRegistry();
+            Resource pcmlFileResource =  registry.getResource(pcmlFileLocation);
+            InputStream pcmlFileContent = pcmlFileResource.getContentStream();
```

Comment:
Input stream close in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/commits/22986634427adb228f51d47163e25fa13cbc10bc